generic/gdb.py

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env python
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2016 IBM
+# Author: Pavithra <pavrampu@linux.vnet.ibm.com>
+
+import os
+import re
+from avocado import Test
+from avocado import main
+from avocado.utils import archive
+from avocado.utils import build
+from avocado.utils import distro
+from avocado.utils import process
+from avocado.utils.software_manager import SoftwareManager
+
+
+class GDB(Test):
+
+    def setUp(self):
+        sm = SoftwareManager()
+        dist = distro.detect()
+        packages = []
+        if dist.name == 'Ubuntu':
+            packages = ['gcc',
+                        'g++',
+                        'dejagnu',
+                        'flex',
+                        'binutils-dev',
+                        'bison']
+        elif dist.name == 'SuSE' or dist.name == 'redhat' or dist.name == 'fedora':
```

Comment:
Yes, good one. And since we are talking about styling, as (I suspect) you're not going to test `if packages` in line 51, you could avoid code duplication by appending only the packages that are different:
```python
...
packages = ['gcc', 'dejagnu', 'flex', 'bison']
if dist.name == 'Ubuntu':
    packages.extend(['g++', 'binutils-dev'])
...
```